From 026707ddf16421d245b23e0cb86b7cc2a0fb26d4 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Fri, 6 Apr 2012 16:27:30 -0700
Subject: [PATCH 037/337] MIPS: OCTEON/MSI: Don't compile unused functions
 when !CONFIG_SMP

Based On SDK 3.0.0-482

octeon_irq_msi_set_affinity_pcie() and
octeon_irq_msi_set_affinity_pci() are unused for !CONFIG_SMP, so don't
build them.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
Signed-off-by: Jack Tan <jack.tan@windriver.com>
---
 arch/mips/pci/msi-octeon.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/arch/mips/pci/msi-octeon.c b/arch/mips/pci/msi-octeon.c
index 93e61e9..0165b14 100644
--- a/arch/mips/pci/msi-octeon.c
+++ b/arch/mips/pci/msi-octeon.c
@@ -291,6 +291,8 @@ static int octeon_irq_irq_to_msi_scatter(int irq)
 	return ((t << 6) & 0xc0) | ((t >> 2) & 0x3f);
 }
 
+#ifdef CONFIG_SMP
+
 static atomic_t affinity_in_progress[4] = {
 	ATOMIC_INIT(1),
 	ATOMIC_INIT(1),
@@ -350,6 +352,7 @@ static int octeon_irq_msi_set_affinity_pci(struct irq_data *data,
 	atomic_add(1, affinity_in_progress + index);
 	return r;
 }
+#endif /* CONFIG_SMP */
 
 static void octeon_irq_msi_enable_pcie(struct irq_data *data)
 {
-- 
1.7.5.4

